### PR TITLE
chore(catalog): export RelatedEntitiesCard component

### PR DESCRIPTION
**Pull request by djamaile**
Signed-off-by: djamaile <rdjamaile@gmail.com>

#### :heavy_check_mark: Checklist

- [x ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by djamaile** [comment]
Wasn't sure about the naming, if you have better name please let me know! I also made the entityKind optional because I want to render the relation of all entities it can find and not just one.

File: plugins/catalog/src/plugin.ts [hunk 1]
```diff
@@ -186,3 +186,15 @@ export const EntitySystemDiagramCard = catalogPlugin.provide(
     },
   }),
 );
+
+export const EntityGenericRelationshipCard = catalogPlugin.provide(
```
Comment (freben) [inline plugins/catalog/src/plugin.ts:190]: Just plain `RelatedEntitiesCard` is fine with me.

Seems you forgot to export it all the way though - I expected to see a change in `index.ts` and a corresponding generated change in `api-report.md`.

Comment (djamaile) [reply plugins/catalog/src/plugin.ts]: Updated it, don’t know why the goalie workflow failed though.

**Comment by freben** [comment]
yah sorry i added a little fixup thing on top as well, because i saw we'd need to change the component to being a const instead of a function and have "hidden" destructuring of the props to look nice. Waiting for it to go green now ...

**Review by freben — APPROVED** [review]

**Comment by djamaile** [comment]
@freben good to know for future contributions, thanks!

**Pull request merged**